query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -173,4 +254,17 @@ public ResultSetMetadata getResultSetMetadata() {
   public LinkedHashSet<Expression> getAllSelections() {
     return this.allSelections;
   }
+
+  public Optional<Double> getTimeSeriesColumn(String columnName) {
+    if (this.timeSeriesColumnMap.containsKey(columnName)) {
+      return Optional.of(Double.parseDouble(this.timeSeriesColumnMap.get(columnName)));
+    }
+    return Optional.empty();
+  }
+
+  public double getTimeFilterDuration(String columnName) {
+    long durationInMillis =
+        this.timeFilterMap.get(columnName).get(1) - this.timeFilterMap.get(columnName).get(0);
+    return (double) durationInMillis / 1000;
```

Comment:
Can we use here `TimeUnit.SECONDS.convert(durationInMillis, TimeUnit.MILLISECONDS)`?